src/features/Configuration/boardControllerConfigSlice.ts

```diff
@@ -66,6 +86,21 @@ const boardControllerConfigSlice = createSlice({
             }: PayloadAction<{ pmicConfigPort: number; configPinState: number }>
         ) {
             state.pmicConfigData.set(pmicConfigPort, configPinState);
+
+            const currentConfig =
+                state.hardwareConfig.pmicPorts?.get(pmicConfigPort);
+
+            // Set the port dirty if..
+            state.pmicConfigDataDirty.set(
+                pmicConfigPort,
+                // ..the current config is not what is stored on the board
+                (currentConfig !== undefined &&
+                    configPinState !== currentConfig) ||
+                    // .. or current config is not the default when no value is stored on the board
+                    (currentConfig === undefined &&
+                        configPinState !==
+                            state.defaultConfig.pmicPorts?.get(pmicConfigPort))
```

Comment:
Maybe extract this om case pf future updates (since it is used 2 places)
Could also be written more verbose then without sacrificing readability maybe.